### PR TITLE
Rules legal update

### PR DESCRIPTION
**Pull request by fsschneider**
Update the rules with the following changes:

- [x] Renames the benchmark/competition to **AlgoPerf**.
- [x] Fixes a typo in the benchmark score formula (was missing the -1 in the normalization).
- [x] Updates the upper integration limit of the benchmark score to 4.
- [x] Updates our terminology to use "fixed" workloads instead of "public workloads".
- [x] Updates our strategy regarding randomized workloads (defining a set of variants, instead of distributions).
- [x] Let submitters provide a per-workload batch size also for the randomized/held-out workloads (and use the same batch size as the fixed workload they are based on by default). 
- [x] Updates the validation/test targets and maximum runtimes according to the repository.
- [x] Increases the number of trials for target setting to $200$ (was $100$).
- [x] Updates our scoring strategy to not directly use held-out workloads (but use them to "disqualify" submissions).

Some open questions regarding the scoring:

- Do we also use held-out workloads for the self-tuning ruleset? Currently written answer: **Yes**.
- Do we also use held-out workloads for the qualification set? Currently written answer: **No**.
- Which submissions do we use to determine "the fastest submission on a held-out workload"? Currently written answer: **Only submissions that successfully train the fixed workload the held-out workload is based on.**

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -35,17 +35,17 @@ We need a more scientifically sound methodology for evaluating training speedups
...
...two tracks: (1) the [Training Algorithm Track](#training-algorithm-track) and (2) the [Model Track](#model-track). The goal of the Training Algorithm TRack is to find training algorithms (optimizers, etc.) that train benchmark models to reach the goal out-of-sample error rate as fast as possible. However, to incentivize practically useful algorithms, in the Training Algorithm Track we require that a single training algorithm simultaneously performs well across all benchmark models and datasets. Similarly, the goal of the Model Track is to find models that can be trained to achieve the target solution quality (out-of-sample error) in the least amount of time on each benchmark dataset. Although submissions in the Model Track will be inherently dataset-specific, we sharply constrain what parts of the training program can be modified in the Model Track and require submitted models to be easily trainable using standard optimizers. Thus the two-track structure discourages overly specific solutions that aren't generally useful to practitioners and will hopefully produce evidence on the relative returns of speeding up training by finding new models or by developing new training algorithms.
```
Comment (georgedahl) [inline RULES.md:38]: capitalization typo "TRack"

File: RULES.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ def get_batch_size(workload_name: str) -> int
 
 - Submitters define a specific batch size for each [workload](#workloads).
 - For example, in advance, they can determine the largest batch size without running out of memory for each workload.
-- For the [randomized workloads](#randomized-workloads), the `workload_name` of the closest public workload will be used in this function.
+- For the [held-out workloads](#randomized-workloads), submitters may provide a batch size once the submission code is frozen and the held-out workloads are sampled from the randomized workloads. By default, this function will use the `workload_name` of the fixed workload it is based on.
```
Comment (georgedahl) [inline RULES.md:145]: Perhaps we should allow submitters to specify batch sizes for all possible held out workloads?

File: RULES.md [hunk 3]
```diff
@@ -361,46 +361,52 @@ Submissions will run on one instance of the [benchmarking hardware](#benchmarkin
...
...n to the [fixed and known workloads](#fixed-workloads), there will also be randomized workloads in our benchmark. These randomized workloads will introduce minor modifications to a fixed workload (e.g. small model changes). The exact instances of these randomized workloads will only be created after the submission deadline and are thus unknown to both the submitters as well as the benchmark organizers. The instructions for creating them, i.e. providing a set or distribution of workloads to sample from, will be defined by this working group and made public with the call for submission, to allow the members of this working group to submit as well as ensure that they do not possess any additional information compared to other submitters. We will refer to the unspecific workloads as *randomized workloads*, e.g. the set or distribution. The specific instance of such a randomized workload we call a *held-out workload*. That is, a held-out workload is a specific sample of a randomized workload that is used for one iteration of the benchmark. While we may reuse randomized workloads between iterations of the benchmark, new held-out workloads will be sampled for each new benchmark iteration.
```
Comment (georgedahl) [inline RULES.md:387]: "call for submission" -> "call for submissions"

File: RULES.md [hunk 4]
```diff
@@ -361,46 +361,52 @@ Submissions will run on one instance of the [benchmarking hardware](#benchmarkin
...
 
-The validation and test target performances on each held-out workload will be defined using the [same protocol as the public workloads](#defining-target-performance) using the same training time budget as the public workload they are based on. To avoid providing "untrainable" held-out workloads, workloads will be rejected if the target-setting procedure on this held-out workload results in a target that is too far away from the public workload. Specifically, if the target of the held-out workload is more than 10% worse relative to the target of its public workload counterpart, it will be rejected. In this case, a new sample is drawn from the randomized workload, and the process of target-setting and checking for rejection is repeated until an acceptable held-out workload is found.
+In the first iteration of this benchmark, we manually design three different workloads variants fo reach fixed workload. The variants are designed such that they achieve a comparable performance to the fixed workload and that they might require different hyperparameters to achieve this performance. After the submission deadline, one held-out workload will be sampled for each fixed workload.
```
Comment (georgedahl) [inline RULES.md:393]: designed

File: RULES.md [hunk 5]
```diff
@@ -444,9 +450,24 @@ An example of a performance profiles plot is shown below, where we plot $\rho_{\
...
+
+For the benchmark score, we compute and integrate the performance profiles using the training times of only the fixed workloads. But we use the submission's performance on the held-out workloads to penalize submissions. Specifically, if a submission is unable to train a held-out workload, we score the submission on the corresponding fixed workload as if that submission did not reach the target. In other words, for a submission to receive a finite training time on a fixed workload, it needs to:
+
+- Reach the validation and test target on the fixed workload within the maximum runtime.
+- Reach the validation and test target fixed workload within 4x of the fastest submission.
+- Reach the validation and test target on the held-out workload (corresponding to the fixed workload) within the maximum runtime.
+- Reach the validation and test target on the held-out workload (corresponding to the fixed workload) within 4x of the fastest submission. To determine the fastest submission on a held-out workload, we only consider submissions that reached the target on the corresponding fixed workload. This protects us against extremely fast but held-out workload-specific submissions.
```
Comment (georgedahl) [inline RULES.md:466]: "This protects us against extremely fast but held-out workload-specific submissions."
->
This protects us against extremely fast submissions that only work on a specific held-out workload-specific and are useless as general algorithms.

**Review by georgedahl — APPROVED** [review]

File: RULES.md [hunk 6]
```diff
@@ -315,12 +315,12 @@ It is also allowed to make the update rule part of the (external) hyperparameter
 </details>
 <br>
 
-It is not allowed to compute any kind of pairwise metrics between the public workloads and the held-out workloads
+It is not allowed to compute any kind of pairwise metrics between the fixed workloads and the held-out workloads
```
Comment (znado) [inline RULES.md:318]: does this need a `.` at the end?

File: RULES.md [hunk 7]
```diff
@@ -361,46 +361,52 @@ Submissions will run on one instance of the [benchmarking hardware](#benchmarkin
...
+Submissions will be scored based on their performance on the [fixed workload](#fixed-workloads). However, additionally submissions must also perform resonably well on a set of [held-out workloads](#randomized-workloads) in order for their score on the fixed workload to count (for full details see the [Scoring](#scoring) section). These held-out workloads will be generated after the submission deadline, but their randomized generating process is publicly available with the call for submission (see "[Randomized workloads](#randomized-workloads)" section).  
 
-#### Public workloads
+Furthermore, a less computationally expensive subset of the fixed workloads is collected with the [qualification set](#qualification-set). Submitters without enough compute ressources to self-report on the full set of fixed and held-out workloads can instead self-report on this smaller qualification set. Well-performing submissions can thereby qualify for computational resources provided by sponsors of the benchmark to be scored on the full benchmark set.
```
Comment (znado) [inline RULES.md:368]: typo: ressources

File: RULES.md [hunk 8]
```diff
@@ -361,46 +361,52 @@ Submissions will run on one instance of the [benchmarking hardware](#benchmarkin
...
 
-The public workloads are fully specified with the call for submissions. They contain a diverse set of tasks such as image classification, machine translation, speech recognition, or other typical machine learning tasks. For a single tasks there might be multiple models and therefore multiple public workloads. The entire set of public workloads should have a combined runtime of roughly 100 hours on the [benchmarking hardware](#benchmarking-hardware).
+#### Fixed workloads
 
-The currently eight public workloads are:
+The fixed workloads are fully specified with the call for submissions. They contain a diverse set of tasks such as image classification, machine translation, speech recognition, or other typical machine learning tasks. For a single tasks there might be multiple models and therefore multiple fixed workloads. The entire set of fixed workloads should have a combined runtime of roughly 100 hours on the [benchmarking hardware](#benchmarking-hardware).
```
Comment (znado) [inline RULES.md:372]: "For a single tasks" -> "For a single task"

File: RULES.md (hunk 4 above)
Comment (znado) [inline RULES.md:393]: "fo reach" -> "for each"

**Review by znado — APPROVED** [review]

**Pull request merged**